web/src/lib/components/shared-components/dialog/dialog.ts

```diff
@@ -0,0 +1,48 @@
+import { writable } from 'svelte/store';
+
+type DialogActions = {
+  onConfirm: () => void;
+  onCancel: () => void;
+};
+
+type DialogOptions = {
+  id: string;
```

Comment:
Requiring an ID isn't strictly needed, because there can only be a single dialog active using the `dialogController`. It would still be beneficial to keep if we want to support multiple dialogs or hide dialogs by ID in the future.